src/plugins/keras_support/switch_operation.cpp

```diff
@@ -0,0 +1,397 @@
+        ir::node_data<std::uint8_t>&& cond,
+        ir::node_data<double>&& then_expr,
+        ir::node_data<double>&& else_expr) const
+    {
+        return primitive_argument_type{
+            cond.scalar() ? std::move(then_expr) : std::move(else_expr)};
+    }
+
+    ///////////////////////////////////////////////////////////////////////////
+    primitive_argument_type switch_operation::switch1d(
+        ir::node_data<std::uint8_t>&& cond,
+        ir::node_data<double>&& then_expr,
+        ir::node_data<double>&& else_expr) const
+    {
+        auto t = then_expr.vector();
+        auto e = else_expr.vector();
+        std::size_t size = t.size();
+
+        ir::node_data<std::uint8_t> condition =
+            extract_value_vector<std::uint8_t>(
+                std::move(cond), size, name_, codename_);
+
+        blaze::UniformVector<bool> ones(size, true);
+        blaze::DynamicVector<bool> c = condition.vector();
+
+        if (!then_expr.is_ref())
+        {
+            t = c * t + (ones - c) * e;
+            return primitive_argument_type{std::move(then_expr)};
+        }
+        else if (!else_expr.is_ref())
+        {
+            e = c * t + (ones - c) * e;
+            return primitive_argument_type{std::move(else_expr)};
+        }
+
+        blaze::DynamicVector<double> result(size);
+
+        result = c * t + (ones - c) * e;
+        return primitive_argument_type{std::move(result)};
+    }
+
+    ///////////////////////////////////////////////////////////////////////////
+    primitive_argument_type switch_operation::switch2d(
+        ir::node_data<std::uint8_t>&& cond,
+        ir::node_data<double>&& then_expr,
+        ir::node_data<double>&& else_expr) const
+    {
+        auto t = then_expr.matrix();
+        auto e = else_expr.matrix();
+        std::size_t rows = t.rows();
+        std::size_t columns = t.columns();
+
+        blaze::DynamicMatrix<bool> c(rows, columns);
+
+        ir::node_data<std::uint8_t> condition;
+        if (cond.num_dimensions() == 2)
+        {
+            condition = extract_value_matrix<std::uint8_t>(
+                std::move(cond), rows, columns, name_, codename_);
+            c = condition.matrix();
+        }
+        else
+        {
+            condition = extract_value_matrix<std::uint8_t>(
+                std::move(cond), columns, rows, name_, codename_);
+            c = blaze::trans(condition.matrix());
+        }
+
+        blaze::UniformMatrix<bool> ones(rows, columns, true);
+
+        if (!then_expr.is_ref())
+        {
+            t = c % t + (ones - c) % e;
+            return primitive_argument_type{std::move(then_expr)};
+        }
+        else if (!else_expr.is_ref())
+        {
+            e = c % t + (ones - c) % e;
+            return primitive_argument_type{std::move(else_expr)};
+        }
+
+        blaze::DynamicMatrix<double> result(rows, columns);
+
+        result = c % t + (ones - c) % e;
```

Comment:
You might want to write this as:
```
blaze::DynamicMatrix<double> result = c % t + (ones - c) % e;
```
to avoid allocating the memory for the result twice (needs verification).